lib/Transforms/IPO/StructFieldAccessInfo.cpp

```diff
@@ -0,0 +1,349 @@
+// lib/Tranforms/IPO/StructFieldAccessInfo.cpp - Implements class of
+// StructFieldAccessInfo -*- C++ -*-===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===------------------------------------------------------------------------===//
+//
+// This file implements StructFieldAccessInfo that is used in
+// StructFieldCacheAnalysis pass.
+//
+//===------------------------------------------------------------------------===//
+
+#include "StructFieldCacheAnalysisImpl.h"
```

Comment:
There are some header files in lib/Analysis/ and lib/IR/ and comments said they are private to some cpp files. (e.g. lib/IR/ConstantsContext.h) I thought it's a good practice. I can move the header file to include/ also